hatch_build.py

```diff
@@ -67,81 +110,314 @@ def install_pact_binaries(self, version: str) -> None:  # noqa: PLR0912
                 machine = "x86_64"
             else:
                 msg = f"Unknown macOS machine {platform}"
-                raise ValueError(msg)
-            url = PACT_URL.format(version=version, os=os, machine=machine, ext="tar.gz")
+                warnings.warn(msg, RuntimeWarning, stacklevel=2)
+                return None
+            return PACT_BIN_URL.format(
+                version=version,
+                os=os,
+                machine=machine,
+                ext="tar.gz",
+            )
 
-        elif platform.startswith("win"):
+        if platform.startswith("win"):
             os = "windows"
 
             if platform.endswith("amd64"):
                 machine = "x86_64"
             elif platform.endswith(("x86", "win32")):
                 machine = "x86"
+            else:
+                msg = f"Unknown Windows machine {platform}"
+                warnings.warn(msg, RuntimeWarning, stacklevel=2)
+                return None
+            return PACT_BIN_URL.format(
+                version=version,
+                os=os,
+                machine=machine,
+                ext="zip",
+            )
+
+        if "linux" in platform and "musl" not in platform:
+            os = "linux"
+            if platform.endswith("x86_64"):
+                machine = "x86_64"
+            elif platform.endswith("aarch64"):
+                machine = "arm64"
+            else:
+                msg = f"Unknown Linux machine {platform}"
+                warnings.warn(msg, RuntimeWarning, stacklevel=2)
+                return None
+            return PACT_BIN_URL.format(
+                version=version,
+                os=os,
+                machine=machine,
+                ext="tar.gz",
+            )
+
+        msg = f"Unknown platform {platform}"
+        warnings.warn(msg, RuntimeWarning, stacklevel=2)
+        return None
+
+    def _pact_bin_extract(self, artifact: Path) -> None:
+        """
+        Extract the Pact binaries.
+
+        The upstream distributables contain a lot of files which are not needed
+        for this library. This function ensures that only the files in
+        `pact/bin` are extracted to avoid unnecessary bloat.
```

Comment:
the files in pact/bin are only wrapper scripts which call the files in the lib directory, which contains the ruby runtime and the actual pact exectuables.

https://github.com/pact-foundation/pact-ruby-standalone/blob/7d6467e942884cf88f171de34bd1a6b5dbe94100/packaging/pact-broker.sh#L29

<img width="1521" alt="Screenshot 2023-10-05 at 19 44 58" src="https://github.com/pact-foundation/pact-python/assets/19932401/f8800dc5-44de-44db-a1f5-5555e2c2582f">

The unit tests are just checking for the wrapper scripts existence.

Reply:
For some reason, I had the idea that Ruby managed to combine the libs into a single executable... That is my mistake!

I'm rather worried now that the unit tests don't actually test the library properly.

Reply:
This has been fixed in 99f7657